modules/nextflow/src/main/groovy/nextflow/scm/AzureDevOpsRepositoryProvider.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020, Seqera Labs
+ * Copyright 2013-2019, Centre for Genomic Regulation (CRG)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.scm
+
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import groovy.transform.Memoized
+
+/**
+ * Implements a repository provider for Azure DevOps service
+ *
+ * @author Tobias Neumann <tobias.neumann.at@gmail.com>
+ */
+@CompileStatic
+final class AzureDevOpsRepositoryProvider extends RepositoryProvider {
+
+    private String user
+    private String repo
+
+    AzureDevOpsRepositoryProvider(String project, ProviderConfig config=null) {
+        this.project = project
+        this.user = this.project.tokenize('/').first()
+        this.repo = this.project.tokenize('/').last()
+        this.config = config ?: new ProviderConfig('azuredevops')
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    String getName() { "Azure DevOps" }
```

Comment:
Let's call it *Azure Repos* 

https://azure.microsoft.com/en-us/services/devops/repos/